client-app/core/composables/useThemeContext.ts

```diff
@@ -1,21 +1,34 @@
 import { computed, ref } from "vue";
-import { IS_DEVELOPMENT } from "../constants";
-import type { IThemeConfig, IThemeContext } from "../types";
+import { getStoreSettings } from "@/core/api/graphql/settings";
+import type { IThemeConfig, IThemeConfigPreset, IThemeContext } from "../types";
+import type { GetStoreSettingsQuery } from "@/core/api/graphql/types";
 
 const themeContext = ref<IThemeContext>();
 
 export function useThemeContext() {
-  async function fetchThemeContext() {
-    const result: IThemeContext = await (await fetch("/storefrontapi/theme/context")).json();
-
-    if (IS_DEVELOPMENT) {
-      // TODO: remove this when switching to SSR
-      const settings = (await import("../../../config/settings_data.json")) as IThemeConfig;
-
-      result.settings = typeof settings.current === "string" ? settings.presets[settings.current] : settings.current;
-    }
-
-    themeContext.value = result;
+  function fetchThemeContext() {
+    let data: GetStoreSettingsQuery["store"];
+    let settings: IThemeConfigPreset;
+
+    const context = getStoreSettings("B2B-store").then((res) => {
+      data = res;
+    });
+    const config = import("../../../config/settings_data.json").then((res) => {
+      const allSettings = res as IThemeConfig;
+      settings =
+        typeof allSettings.current === "string" ? allSettings.presets[allSettings.current] : allSettings.current;
+    });
+
+    return Promise.all([config, context]).then(() => {
+      if (!data) {
+        return;
+      }
+      themeContext.value = {
+        ...data,
+        settings,
+        settingsFromPlatform: data.settings,
+      };
+    });
```

Comment:
```typescript
const [store, allSettings] = await Promise.all([getStoreSettings(STORE_NAME), import("../../../config/settings_data.json")]);
const currentSettings = typeof allSettings.current === "string" ? allSettings.presets[allSettings.current] : allSettings.current;
themeContext.value = {
  ...store,
  ...currentSettings,
};
```

And why do we need `settingsFromPlatform`"? What purpose?

Reply:
as a marker for theme_settings / store_settings confuse